torch/_guards.py

```diff
@@ -343,3 +343,27 @@ def tracing(context: TracingContext):
         yield _CURRENT_TRACING_CONTEXT
     finally:
         _CURRENT_TRACING_CONTEXT = old_context
+
+
+# Subclasses can be found in torch/_dynamo/source.py
+@dataclasses.dataclass
+class Source:
+    def reconstruct(self, codegen):
+        raise NotImplementedError()
+
+    def guard_source(self):
+        raise NotImplementedError()
+
+    def name(self):
+        raise NotImplementedError()
+
+    def make_guard(self, fn, is_volatile=False):
+        if self.guard_source() is GuardSource.CONSTANT:
+            raise NotImplementedError()
```

Comment:
Is this correct?

Reply:
I mean, it's your code!! lol. I would hope you know.

The constant source stuff is pretty mushy for me. But it at least sort of makes sense that you shouldn't ever be guarding on them; they're constant after all, why would you guard on something that the user has said should never change from iteration to iteration.